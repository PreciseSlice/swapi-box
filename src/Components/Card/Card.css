.card {
  align-items: center;
  border: 1px solid #feda4a;
  color: #feda4a;
  display: flex;
  margin-top: 20px;
  justify-content: center;
  width: 20%; }
  .card h3 {
    justify-content: center; }
