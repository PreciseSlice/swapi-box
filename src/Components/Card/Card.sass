.card 
  align-items: center
  border: 1px solid #feda4a
  color: #feda4a
  display: flex
  flex-direction: column
  height: 100%
  margin-top: 30px
  justify-content: center
  width: 30%

  ul 
    margin: 9% 3%
    line-height: 2
    font-size: 1.2em

  button
    background-color: #4b074b
    border: none
    box-shadow: 0 0 5px 3px rgba(0, 59, 168, 0.8)
    color: #feda4a
    font-family: 'Oswald', sans-serif
    font-size: 1.6em
    height: 60px
    letter-spacing: 0.15em
    margin: 20px
    outline: none
    width: 60%
    &:hover
      background-color: #feda4a
      color: #4b074b

  h1 
    letter-spacing: 0.15em
    font-size: 2.2em
    font-weight: 600
    margin-top: 3%


.highlighted 
  
  button
    background-color: #feda4a
    color: #4b074b
    box-shadow: 0 0 5px 3px rgba(0, 59, 168, 0.8)


@media (max-width: 1050px)
  .card 
    width: 40%
    
    h1 
      font-size: 1.8em
    
    ul
      font-size: 1em

@media (max-width: 775px)
  .card 
    width: 80%

@media (max-width: 500px)
  .card 
    width: 96%

@media (max-width: 350px)
  .card h1
    font-size: 1.5em