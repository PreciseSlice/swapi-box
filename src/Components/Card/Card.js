import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';

const Card = ({ person }) => {
  return (
    <div className="card">
      {person && (
        <div className="personCard">
          <h1>{person.name}</h1>
          <div>
            <h3>{`Homeworld: ${person.homeworld}`}</h3>
            <h3>{`Population: ${person.population}`}</h3>
            <h3>{`Species: ${person.speciesName}`}</h3>
            {/* favorite button */}
          </div>
        </div>
      )}
    </div>
  );
};

export default Card;

Card.propTypes = {
  person: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      homeWolrd: PropTypes.string.isRequired,
      population: PropTypes.number.isRequired,
      speciesName: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
};
