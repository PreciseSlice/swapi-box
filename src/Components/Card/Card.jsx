import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';

const Card = ({ data, clickHandler }) => {
  const dataKeys = Object.keys(data).filter(keys => keys !== 'name');

  const keyMap = dataKeys.map((value, index) => {
    return <li key={`${Date.now}${index}`}>{`${value}: ${data[value]}`}</li>;
  });

  return (
    <div className="card">
      <h1>{data.name}</h1>
      <ul>{keyMap}</ul>
      <button className="favorite-button" onClick={() => clickHandler(data)}>
        Favorite
      </button>
    </div>
  );
};

export default Card;

Card.propTypes = {
  data: PropTypes.Object
}.isRequired;
