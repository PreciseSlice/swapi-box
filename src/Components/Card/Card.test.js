/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card.jsx';
import { shallow, mount, render } from 'enzyme';

describe('Card', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card
        data={{
          name: 'Alderaan',
          climate: 'temperate',
          terrain: 'grasslands, mountains',
          population: '2000000000'
        }}
      />
    );
  });

  it('Exist and matches snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should take all data except name: passed into it and make li items using the key value pairs', () => {
    expect(wrapper.find('li').length).toEqual(3);
    expect(wrapper.find('li').first().text()).toEqual('climate: temperate')
    expect(wrapper.find('li').last().text()).toEqual('population: 2000000000')
  });
});
