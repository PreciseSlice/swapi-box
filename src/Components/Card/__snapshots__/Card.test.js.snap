// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Exist and matches snapshot 1`] = `
<div
  className="card undefined"
>
  <h1>
    Alderaan
  </h1>
  <ul>
    <li
      key="function now() { [native code] }0"
    >
      climate: temperate
    </li>
    <li
      key="function now() { [native code] }1"
    >
      terrain: grasslands, mountains
    </li>
    <li
      key="function now() { [native code] }2"
    >
      population: 2000000000
    </li>
  </ul>
  <button
    onClick={[Function]}
  >
    favorite
  </button>
</div>
`;
